test/python/sql_validation.py

```diff
@@ -0,0 +1,402 @@
+    for i in files:
+        with open(i, "r") as testfile:
+            testline = testfile.readline()
+
+            while testline:
+                testline = testline.strip()
+
+                # ignoring comments in the file
+                if testline.startswith("--") or testline.startswith("/*") or testline.startswith("#"):
+                    testline = testfile.readline()
+                    continue
+
+                # if pattern exists, return true
+                elif(re.search( pattern, testline, re.I)):
+                    return True
+                testline = testfile.readline()
+    return False
+
+
+# find the object name in the test files
+def find_in_JDBC(fname, logger):
+
+    # get the list of JDBC input files
+    files = list_files("../JDBC/input", "*.*")
+
+    # get the set of create object_name
+    object_name = find_pattern_create(logger)
+    
+    # set output file path
+    path = Path.cwd().joinpath("output", "sql_validation_framework")
+    Path.mkdir(path, parents = True, exist_ok = True)
+    f_path = path.joinpath(fname + ".out")
+
+    with open(f_path, "w") as expected_file:
+        for object in object_name:
+
+            # flag for object name found or not in the JDBC input files
+            flag = False
+
+            pattern = object[1]
+
+            # replacing . as in regex it means any character
+            if "." in object[1]:
+                pattern = object[1].replace(".", "[.]")
+
+            # adding word boundary if object is not a function
+            if not object[0] == 'function':
+                pattern = pattern + r"\b"
+
+            flag = find_obj(files, r"\b" + pattern)
+
+            # searching again without schema name for objects in sys schema
+
+            if(flag == False and "sys." in object[1]):
+                pattern = r"\b" + pattern.replace("sys[.]", "")
+                
+                # creating pattern to identify @@func calls
+                if object[0] == 'function':
+                    pattern = "(" + "@@" + pattern.replace("\s{0,2}[(]", "").replace("\\b", "") + ")|(" +  pattern + ")"
+                
+
+                flag = find_obj(files, pattern)
+            
+            # if tests not found, add it to output file
+            if flag == False:
+                expected_file.write("Could not find tests for {0} {1}\n".format(object[0], object[1].split('\s')[0]))
+    logger.info("Tests for objects found successfully!")
+
+
+
+# find patterns in the framework
+def find_patterns(logfname, logger):
+    
+    logger.info("Running tests for pattern search!!")
+    
+    # get the list of patterns from config
+    patterns = cfg["searchPatterns"].split(',')
+    found = True
+    result = False
+
+    for pattern in patterns:
+        # generate outputfile name
+        filename = "expected_" + re.sub("[^a-zA-Z0-9 ]", "", pattern)
+        
+        # search for the pattern
+        find_pattern_drop(pattern, filename, logger)
```

Comment:
I think we should rename this function `find_pattern` or similar since this function is being used for multiple patterns.

Reply:
Updated